CircleCI cannot build the docker image after the upgrade to alpine 3.14

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks! -->

**Describe the bug**

After the base docker image has been upgraded by https://github.com/falcosecurity/event-generator/pull/49, the [CircleCI release job](https://app.circleci.com/pipelines/github/falcosecurity/event-generator/86/workflows/2df3bc9d-d975-43ec-ad02-7bc4bdd1bdce/jobs/108) is failing due to the `faccessat2` issue described in the [Release Notes for Alpine 3.14.0](https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.14.0#faccessat2).

**How to reproduce it**

The same docker and libseccomp versions used by CircleCI are needed to reproduce the problem. In our case:
```
circleci@0cdbdda9a9f2:~/project$ docker version
Client: Docker Engine - Community
 Version:           19.03.12
 API version:       1.32
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:42:53 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          17.09.0-ce
  API version:      1.32 (minimum version 1.12)
  Go version:       go1.8.3
  Git commit:       afdb6d4
  Built:            Tue Sep 26 22:40:56 2017
  OS/Arch:          linux/amd64
  Experimental:     false
```

Then, in such an environment, run `make image`.


**Expected behaviour**

No failure.

**Screenshots**

```
circleci@0cdbdda9a9f2:~/project$ make image
docker build \
	-t "docker.io/falcosecurity/event-generator:HEAD" \
	-f Dockerfile .
Sending build context to Docker daemon  631.3kB
Step 1/11 : FROM golang:1.16.5-alpine3.14 as builder
 ---> 2592d553708f
Step 2/11 : LABEL maintainer "cncf-falco-dev@lists.cncf.io"
 ---> Using cache
 ---> 2159dd4e3b61
Step 3/11 : RUN apk add --no-cache make bash git build-base
 ---> Using cache
 ---> 06daabef3e4a
Step 4/11 : WORKDIR /event-generator
 ---> Using cache
 ---> de4731ff50a8
Step 5/11 : COPY . .
 ---> ab7d992da146
Step 6/11 : RUN make
 ---> Running in 8a11f0f8ebf5
rm -f -R event-generator
make: /bin/bash: Operation not permitted
make: /bin/bash: Operation not permitted
make: /bin/bash: Operation not permitted
make: /bin/bash: Operation not permitted
make: /bin/bash: Operation not permitted
make: *** [Makefile:34: clean] Error 127
The command '/bin/sh -c make' returned a non-zero code: 2
make: *** [Makefile:58: image] Error 2
```

<!-- If applicable, add screenshots to help explain your problem. -->

**Environment**

<!-- Please complete the following info. -->

- Falco version:
<!-- Use "falco --version". -->
- System info:
<!-- Falco has a built-in support command you can use  "falco --support | jq .system_info" -->
- Cloud provider or hardware configuration:
- OS:
<!-- Eg., output of "cat /etc/os-release". -->
- Kernel:
<!-- Eg., output of "uname -a". -->
- Installation method:
<!-- Eg., Kubernetes, RPM, DEB, from source? -->

**Additional context**

This issue is blocking the release of [v0.6.2](https://github.com/falcosecurity/event-generator/milestone/10). I'm going to delete the git tag, and release it again once this is fixed.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -22,7 +22,7 @@ workflows:
 jobs:
   test:
     docker:
-      - image: docker.io/golang:1.16.5
+      - image: docker.io/golang:1.16.6
     steps:
       - checkout
       - run:
@@ -30,18 +30,21 @@ jobs:
           command: make test
   build:
     docker:
-      - image: docker.io/golang:1.16.5
+      - image: docker.io/circleci/golang:1.16.6
     steps:
       - checkout
+      - setup_remote_docker:
+          version: 20.10.6
       - run:
           name: Build
           command: make build
   release:
     docker:
-      - image: circleci/golang:1.16.5
+      - image: docker.io/circleci/golang:1.16.6
     steps:
       - checkout
-      - setup_remote_docker
+      - setup_remote_docker:
+          version: 20.10.6
       - run:
           name: Prepare env
           command: |

